src/main/java/echoquery/sql/QueryBuilder.java

```diff
@@ -0,0 +1,87 @@
+package echoquery.sql;
+
+import java.util.Optional;
+
+import com.amazon.speech.slu.Intent;
+import com.facebook.presto.sql.QueryUtil;
+import com.facebook.presto.sql.tree.ComparisonExpression;
+import com.facebook.presto.sql.tree.Expression;
+import com.facebook.presto.sql.tree.QualifiedName;
+import com.facebook.presto.sql.tree.Query;
+import com.facebook.presto.sql.tree.Relation;
+import com.facebook.presto.sql.tree.Select;
+import com.facebook.presto.sql.tree.StringLiteral;
+
+import echoquery.utils.SlotNames;
+
+public class QueryBuilder {
+
+  private Select select;
+  private Relation from;
+  private Optional<Expression> where;
+  
+  public QueryBuilder() {}
+
+  public QueryBuilder(Select select, Relation from) {
+    this.select = select;
+    this.from = from;
+    this.where = Optional.empty();
+  }
+
+  public QueryBuilder select(Select select) {
+    this.select = select;
+    return this;
+  }
+
+  public QueryBuilder from(Relation from) {
+    this.from = from;
+    return this;
+  }
+
+  public QueryBuilder where(Expression where) {
+    this.where = Optional.of(where);
+    return this;
+  }
+
+  public Query build() {
+    if (where != null && where.isPresent()) {
+      return QueryUtil.simpleQuery(select, from, where.get());
+    } else {
+      return QueryUtil.simpleQuery(select, from);
+    }
+  }
+
+  public static QueryBuilder from(Intent intent) {
+    String table = intent.getSlot(SlotNames.TABLE_NAME).getValue();
+
+    QueryBuilder builder = new QueryBuilder()
+        .select(QueryUtil.selectList(QueryUtil.functionCall("COUNT")))
```

Comment:
Hmm I think I did this at first because I thought constructor would couple QueryBuilder too close to how we're implementing it, but I think this could be a constructor yeah